### PR TITLE
[OTHER] - Add path aliases (#3564)

### LINKED ISSUE #3564: [OTHER] Suggestion - Add path aliases
### What would you like to share?

Right now, we are having to manually specify the directory path in this way:

`import xyz from "../../../components/XYZComponent"
`

With aliasing using jsconfig.json file, we can clean up the import syntax, and do something like,

`import xyz from 'components/XYZComponent'
`

Next.js supports it out of the box: https://nextjs.org/docs/advanced-features/module-path-aliases

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by FrozenHearth**
## Fixes Issue

Closes #3564 

## Changes proposed

Right now, we are having to manually specify the directory path in this way:

import xyz from `../../../components/XYZComponent`

With aliasing using jsconfig.json file, we can clean up the import syntax, and do something like,

import xyz from `@/components/XYZComponent `

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

N/A

## Note to reviewers

**I haven't made the changes in mdx files. Please let me know if I need to.**

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/4543"></a>

### CODE REVIEW

**Review by priyankarpal — APPROVED** [review]
Cool.

File: jsconfig.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "compilerOptions": {
+        "baseUrl": ".",
+        "paths": {
+            "@/components/*": [
+                "components/*"
+            ]
```
Comment (FrozenHearth) [inline jsconfig.json:7]: We can add path aliases for config and other folders later on. This is customizable.

**Review by priyankarpal — CHANGES_REQUESTED** [review]
hey @FrozenHearth this branch has a conflict error please resolve it.

**Comment by FrozenHearth** [comment]
> hey @FrozenHearth this branch has a conflict error please resolve it.

Thanks for the review. Resolved the conflict.

**Review by priyankarpal — APPROVED** [review]
Now it's looking good. 🎉

File: jsconfig.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+    "compilerOptions": {
+        "baseUrl": ".",
+        "paths": {
+            "@/components/*": [
```
Comment (eddiejaoude) [inline jsconfig.json:5]: I think the common style is to use `@component` without the `/`? 

Which would create the path `import Link from "@components/Link";

Comment (FrozenHearth) [reply jsconfig.json:5]: If you want, I can make that change and push. Please let me know if I should make it.

Comment (FrozenHearth) [reply jsconfig.json]: Got it, changes done. Thanks.

Comment (FrozenHearth) [reply jsconfig.json]: Conflicts also resolved.

Comment (FrozenHearth) [reply jsconfig.json]: Additional conflicts resolved.

Comment (eddiejaoude) [reply jsconfig.json]: Thank you 👍

**Comment by eddiejaoude** [comment]
Looks good 👍 I will resolve merge conflicts when testing and merging

**Comment by FrozenHearth** [comment]
Thank you very much! @eddiejaoude

File: jsconfig.json (hunk 1 above)
Comment (eddiejaoude) [reply jsconfig.json]: Great idea, thank you

**Pull request merged**